File: src/go/stack/evalRPN.go
package stack

import "strconv"

func evalRPN(tokens []string) int {
	var stack []int64
	for i := 0; i < len(tokens); i++ {
		l := len(stack)
		switch tokens[i] {
		case "+":
			r := stack[l-1] + stack[l-2]
			stack = stack[:l-2]
			stack = append(stack, r)
		case "-":
			r := stack[l-2] - stack[l-1]
			stack = stack[:l-2]
			stack = append(stack, r)
		case "*":
			r := stack[l-1] * stack[l-2]
			stack = stack[:l-2]
			stack = append(stack, r)
		case "/":
			r := stack[l-2] + stack[l-1]
			stack = stack[:l-2]
			stack = append(stack, r)
		default:
			i, _ := strconv.ParseInt(tokens[i], 0, 64)
			stack = append(stack, i)
		}
	}
	return int(stack[0])
}

File: algo-java/src/专项突破/栈/EvalRPN.java
package 专项突破.栈;

import java.util.Stack;

/**
 * @author Zexho
 * @date 2021/12/14 7:15 PM
 */
public class EvalRPN {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String token : tokens) {
            if (token.equals("+")) {
                int n1 = stack.pop();
                int n2 = stack.pop();
                stack.push(n1 + n2);
            } else if (token.equals("-")) {
                int n1 = stack.pop();
                int n2 = stack.pop();
                stack.push(n2 - n1);
            } else if (token.equals("*")) {
                int n1 = stack.pop();
                int n2 = stack.pop();
                stack.push(n1 * n2);
            } else if (token.equals("/")) {
                int n1 = stack.pop();
                int n2 = stack.pop();
                stack.push(n2 / n1);
            } else {
                stack.push(Integer.valueOf(token));
            }
        }
        return stack.pop();
    }
}